### PR TITLE
[OYPD-259] Alerts implementation.

### PR DESCRIPTION
**Pull request by Kerby70**
Jira https://propeople-us.atlassian.net/browse/OYPD-259
D.o https://www.drupal.org/node/2853585

Note: Some styles still need to be addressed within the theme.
Steps to review:
- [x] Go to /node/add
- [x] Add Alert "Global"
  - [x] Fill required fields
    - [x] Select one placement location
  - [x] Leave Reference empty (for global message)
- [x] Go to homepage
- [x] See alert with message with the appropriate background color (Font colors are still being overridden by the theme)
  - [x] See it is only in the selected location
- [x] Repeat with the other location to see only in that selected location.
- [x] Add Alert "Local"
  - [x] Fill required fields
    - [x] Select both placement locations
  - [x] For Reference select OpenY (the homepage)
- [x] Go to homepage
- [x] In header & footer see alert with message with the appropriate background color (Font colors are still being overridden by the theme)
- [x] Go to /blog
- [x] See the first "Global" Alert but not the "Local" Alert referencing Alert
- [x] Click the x on the right
- [x] See alert has disappeared
- [x] Refresh page
- [x] See alert is still not visible (a cookie value has been set per alert/position)

### CODE REVIEW

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============

--------------
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/494/

**Build Log**
*last 25 lines*

 ```
[...truncated 252 lines...]
stdout: You are about to create a /var/www/build494/sites/default/settings.php file and CREATE the 'drupal494' database. Do you want to continue? (y/n): y

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/var/lib/jenkins/reinstall.retry

localhost                  : ok=12   changed=4    unreachable=0    failed=1   

FATAL: command execution failed
hudson.AbortException: Ansible playbook execution failed
	at org.jenkinsci.plugins.ansible.AnsiblePlaybookBuilder.perform(AnsiblePlaybookBuilder.java:227)
	at org.jenkinsci.plugins.ansible.AnsiblePlaybookBuilder.perform(AnsiblePlaybookBuilder.java:200)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:75)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.Build$BuildExecution.build(Build.java:205)
	at hudson.model.Build$BuildExecution.doRun(Build.java:162)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:537)
	at hudson.model.Run.execute(Run.java:1741)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:381)
ERROR: Ansible playbook execution failed
```

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build495

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/495/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build495/profiles/openy/build/reports/behat

**Comment by Sanchiz** [comment]
retest this please

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build502

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/502/

File: modules/openy_features/openy_node/modules/openy_node_alert/config/install/field.field.node.alert.field_alert_belongs.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    - node.type.blog
+    - node.type.branch
+    - node.type.camp
+    - node.type.class
+    - node.type.facility
+    - node.type.landing_page
+    - node.type.membership
+    - node.type.program
+    - node.type.program_subcategory
+    - node.type.session
```
Comment (alexschedrov) [inline modules/openy_features/openy_node/modules/openy_node_alert/config/install/field.field.node.alert.field_alert_belongs.yml]: Can we get rid of these dependencies and put wildcard?

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build502/profiles/openy/build/reports/behat

File: modules/openy_features/openy_node/modules/openy_node_alert/config/install/field.field.node.alert.field_alert_belongs.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      blog: blog
+      branch: branch
+      camp: camp
+      class: class
+      facility: facility
+      landing_page: landing_page
+      membership: membership
+      program: program
+      program_subcategory: program_subcategory
+      session: session
```
Comment (alexschedrov) [inline modules/openy_features/openy_node/modules/openy_node_alert/config/install/field.field.node.alert.field_alert_belongs.yml]: Let's set empty to support all bundles, even when they are created.

**Review by alexschedrov — CHANGES_REQUESTED** [review]

**Comment by alexschedrov** [comment]
1. There's typo in documentation for placement link. It should support only 1 value. Please fix and update documentation.
image

2. Please clean up alert full page display. Right now it looks a bit cluttered and renders referenced entity.
image

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build527

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/527/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build527/profiles/openy/build/reports/behat

**Comment by Kerby70** [comment]
All changes in latest commit.

The cleaned up default display:
image

**Comment by Sanchiz** [comment]
retest this please

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build534

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/534/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build534/profiles/openy/build/reports/behat

File: themes/openy_themes/openy_rose/templates/node/node--alert--full.html.twig [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+{% set is_header = (field_alert_place['#items'].getString() == 'header') %}
+{% if is_header %}
+  {% include 'node--alert--node-alert-header.html.twig' %}
+{% else %}
+  {% include 'node--alert--node-alert-footer.html.twig' %}
+{% endif %}
+<ul>
+  {{ node.field_alert_belongs.value }}
+  {% if node.field_alert_belongs.value is not empty %}
+    <li>Rendered on {{ content.field_alert_belongs }}</li>
```
Comment (alexschedrov) [inline themes/openy_themes/openy_rose/templates/node/node--alert--full.html.twig]: Should be translatable.

File: themes/openy_themes/openy_rose/templates/node/node--alert--full.html.twig [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  {% else %}
+    <li>Rendered on all pages</li>
```
Comment (alexschedrov) [inline themes/openy_themes/openy_rose/templates/node/node--alert--full.html.twig]: Should be translatable.

File: themes/openy_themes/openy_rose/templates/node/node--alert--full.html.twig [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  {% endif %}
+  {% if is_header %}
+    <li>Displayed in the header.</li>
```
Comment (alexschedrov) [inline themes/openy_themes/openy_rose/templates/node/node--alert--full.html.twig]: Should be translatable.

File: themes/openy_themes/openy_rose/templates/node/node--alert--full.html.twig [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  {% else %}
+    <li>Displayed in the footer.</li>
```
Comment (alexschedrov) [inline themes/openy_themes/openy_rose/templates/node/node--alert--full.html.twig]: Should be translatable.

**Review by alexschedrov — CHANGES_REQUESTED** [review]

**Comment by alexschedrov** [comment]
# A few alerts in the same region

When there are a few alerts in the region, they should have pager + slider on mobile.

Current OpenY version:
image


YGS version:
image


The same in footer(on YGS it displays only latest alert, but in distribution should display all with sliders).

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============

--------------
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/545/

**Build Log**
*last 25 lines*

 ```
[...truncated 238 lines...]
stdout: You are about to create a /var/www/build545/sites/default/settings.php file and CREATE the 'drupal545' database. Do you want to continue? (y/n): y

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/var/lib/jenkins/reinstall.retry

localhost                  : ok=12   changed=5    unreachable=0    failed=1   

FATAL: command execution failed
hudson.AbortException: Ansible playbook execution failed
	at org.jenkinsci.plugins.ansible.AnsiblePlaybookBuilder.perform(AnsiblePlaybookBuilder.java:227)
	at org.jenkinsci.plugins.ansible.AnsiblePlaybookBuilder.perform(AnsiblePlaybookBuilder.java:200)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:75)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.Build$BuildExecution.build(Build.java:205)
	at hudson.model.Build$BuildExecution.doRun(Build.java:162)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:537)
	at hudson.model.Run.execute(Run.java:1741)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:381)
ERROR: Ansible playbook execution failed
```

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build547

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/547/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build547/profiles/openy/build/reports/behat

**Comment by Sanchiz** [comment]
retest this please

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build558

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/558/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build558/profiles/openy/build/reports/behat

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build559

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/559/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build559/profiles/openy/build/reports/behat

**Comment by Sanchiz** [comment]
retest this please

**Comment by bjwilkins** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build566

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/566/

**Comment by bjwilkins** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build566/profiles/openy/build/reports/behat

**Review by alexschedrov — APPROVED** [review]
Approved, but we have to write code to support upgrade path.

**Pull request merged**